hub/core/dataset/dataset.py

```diff
@@ -378,6 +369,29 @@ def _load_version_info(self):
         version_state["full_tensors"] = {}  # keeps track of the full unindexed tensors
         self.version_state = version_state
 
+    def _lock(self):
+        storage = get_base_storage(self.storage)
+        if (
+            isinstance(storage, S3Provider)
+            and self.index.is_trivial()
+            and (not self.read_only or self._locked_out)
+        ):
```

Comment:
We are not locking other providers ?

Reply:
Agreed, at the very least we should lock GCS.